TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -1783,4 +1790,27 @@ private static DataDefinition GetDataDefinitionType(Node node, Node symbol, bool
         }
 
     }
+
+    static class RepositoryChecker
+    {
+        // Intrinsic functions which are not allowed in the REPOSITORY paragraph
+        private static readonly HashSet<string> _NotAllowedIntrinsicFunctions = new(StringComparer.OrdinalIgnoreCase){ "WHEN-COMPILED" };
+
+        public static void OnNode([NotNull] Repository repository)
+        {
+            List<SymbolDefinitionOrReference> intrinsicFunctions = repository.CodeElement.IntrinsicFunctions;
+            if (intrinsicFunctions != null)
+            {
+                foreach (var intrinsicFunction in intrinsicFunctions)
+                {
+                    Token token = intrinsicFunction.NameLiteral.Token;
+                    var intrinsicFunctionName = token.SourceText;
```

Comment:
use `instrinicFunction.Name` instead of SourceText

Reply:
Not relevant anymore

Reply:
Done in the new `CodeElementChecker`